src/Sarif.Sarifer/BackgroundAnalysisService.cs

```diff
@@ -23,12 +25,14 @@ internal class BackgroundAnalysisService : IBackgroundAnalysisService
 #pragma warning restore CS0649
 
         /// <inheritdoc/>
-        public async Task StartAnalysisAsync(string path, string text)
+        public async Task StartAnalysisAsync(string path, string text, CancellationToken cancellationToken)
         {
-            foreach (IBackgroundAnalyzer analyzer in this.analyzers)
-            {
-                await analyzer.StartAnalysisAsync(path, text).ConfigureAwait(continueOnCapturedContext: false);
-            }
+            Task[] tasks = this.analyzers
+                .Select(async a => await a.StartAnalysisAsync(path, text, cancellationToken).ConfigureAwait(continueOnCapturedContext: false))
```

Comment:
>async [](start = 24, length = 5)

remove async, and await and configurate await from this line.

this should really be just

return Task.WhenAll(this.analyzers.Select(analyzer => analyzer.StartAnalysisAsnc(path, text cancellationToken);

short sweet one-liner